### ISSUE TITLE
Doesn't seem to work on AMD

### ISSUE THREAD

**Issue by antonov-a** [zRank/EthVeoDualMiner#1]
Hello!

I've tested AMD miner on a following setup:
AMDGPU-pro 17.40
Kernel 4.4
AMD Radeon RX 580

And it dies right after launch.
Here's the last lines of strace that is shown before that:
```
nanosleep({1, 0}, NULL) = 0
nanosleep({1, 0}, <unfinished ...>
+++ killed by SIGSEGV (core dumped) +++
Segmentation fault (core dumped)
```
NVI seems to work fine, though. I'll check pool side once there's more data.

**Comment by zRank** [comment]
Yes, the RX 470 and RX 580s seem to crash. I've been working to identify where and why. There's something in the kernel code that crashes on Program Build call rather than returning errors, so it has been very difficult to track down exactly why this happens.

Other AMD cards like the Vega 56 and 64 do work.

**Comment by tallakt** [comment]
I would also like R4/5x0
